### DISCUSSION TITLE
SEQUENCER compatible with Ableton 10 ?

### DISCUSSION THREAD

**Discussion by Zedrock** [raphaelquast/beatstep#59] [Q&A]
Hello,
Woh! Such a great sequencer you made here!
Do you think it will be possible to have the sequencer, even less elaborated, that could work with LIVE 10 ?
Unless it is based on features that are only accessible under LIVE 11 ?

Best regards

Zed'

**Comment by raphaelquast** [comment]
Hey, thanks 😄 
well... it's certainly possible to make the sequencer compatible with ableton 10...
As far as I know, the only ableton-11 specific functionalities I've implemented  are "note-probability" and "velocity-deviation" ... the rest should be just fine in ableton 10.

The actual problem is that the python API for creating and editing MIDI-notes has changed a lot between 10 and 11...
So in order to make this work one would have to do 2 things:
- adjust the functions that initialize and update MIDI notes to the Ableton 10 API
- make sure no Ableton 11 specific functions (and note-properties) are accessed from Ableton 10

However, since I am using Ableton 11, implementing this is not on my priority list right now... 
... but I'd be happy to have a look at pull-requests that try to make the sequencer compatible with Ableton 10!
